aspnetcore/fundamentals/http-requests.md

```diff
@@ -0,0 +1,392 @@
+
+    services.AddMvc();
+}
+```
+
+The defined interface can then be consumed wherever necessary in code, with the implementation provided by the DI framework and Refit.
+
+```csharp
+[ApiController]
+public class ValuesController : ControllerBase
+{
+    private readonly IHelloClient _client;
+
+    public ValuesController(IHelloClient client)
+    {
+        _client = client;
+    }
+    
+    [HttpGet("/")]
+    public async Task<ActionResult<Reply>> Index()
+    {
+        return await _client.GetMessageAsync();
+    }
+}
+```
+
+## Outgoing request middleware
+
+`HttpClient` already has the concept of delegating handlers that can be linked together for outgoing HTTP requests. The `HttpClientFactory` makes registering of these per named client more intuitive. It supports registering and chaining multiple handlers together to build an outgoing request middleware pipeline. Each of these handlers is able to perform work before and after the outgoing request, in a very similar pattern to the inbound middleware pipeline in ASP.NET Core. This provides a mechanism to manage cross cutting concerns around HTTP requests, including caching, error handling, serialization and logging for example.
+
+To create a handler, a class can be added, deriving from `DelegatingHandler`. The `SendAsync` method can then be overridden to execute code before passing the request to the next handler in the pipeline.
+
+```csharp
+private class RetryHandler : DelegatingHandler
+{
+    public int RetryCount { get; set; } = 5;
+
+    protected override Task<HttpResponseMessage> SendAsync(HttpRequestMessage request, CancellationToken cancellationToken)
+    {
+        for (var i = 0; i < RetryCount; i++)
+        {
+            try
+            {
+                return base.SendAsync(request, cancellationToken);
+            }
+            catch (HttpRequestException) when (i == RetryCount - 1)
+            {
+                throw;
+            }
+            catch (HttpRequestException)
+            {
+                // Retry
+                Task.Delay(TimeSpan.FromMilliseconds(50));
+            }
+        }
+
+        // Unreachable.
+        throw null;
+    }
+}
+```
+
+The preceding code, defines a basic retry handler which retries up to five times if a `HttpRequestException` is caught.
+
+During registration, one or more handlers can be added to the configuration for a `HttpClient` via extension methods on the `HttpClientBuilder`.
+
+```csharp
+public static void Configure(IServiceCollection services)
+{
+    services.AddTransient<RetryHandler>();
+
+    services.AddHttpClient("example", c =>
+    {
+        c.BaseAddress = new Uri("https://localhost:5000/");
+    })
+    .AddHttpMessageHandler<RetryHandler>(); // Retry requests to GitHub using the retry handler
+}
+```
+
+Here the `RetryHandler` is registered as a transient service with the DI framework. Once registered `AddHttpMessageHandler` can be called, passing in the type for the handler.
```

Comment:
- Here --> In the preceding code,
- the DI framework --> DI
- Add a comma after "Once registered"